src/compiler/tracing.ts

```diff
@@ -1,16 +1,25 @@
+/* Tracing events for the compiler. */
+
 /*@internal*/
-/** Tracing events for the compiler. */
-namespace ts.tracing {
+namespace ts { // eslint-disable-line one-namespace-per-file
+    // should be used as tracing?.___
+    export let tracing: typeof tracingEnabled | undefined;
+    // enable the above using startTracing()
+}
+
+// `tracingEnabled` should never be used directly, only through the above
+/* @internal */
+namespace ts.tracingEnabled { // eslint-disable-line one-namespace-per-file

```

Comment:
We could have just declared `tracingEnabled` as a non-exported namesapce in `ts` (similar to the `Parser` namespace in _parser.ts_`). Do we ever need to reference `ts.tracingEnabled` directly outside of `ts.startTracing`?

Reply:
I don't know how I missed it...